src/plugins_types/union.c

```diff
@@ -228,6 +256,11 @@ union_store_type(const struct ly_ctx *ctx, struct lysc_type_union *type_u, uint3
     if ((rc != LY_SUCCESS) && (rc != LY_EINCOMPLETE)) {
         /* clear any leftover/freed garbage */
         memset(&subvalue->value, 0, sizeof subvalue->value);
+
+        /* if this is a leafref, lets make sure we propagate the appropriate error, and not a type validation failure */
+        if (type->basetype == LY_TYPE_LEAFREF) {
```

Comment:
Redundant check, since it is in the function.

Reply:
done